packages/@uppy/aws-s3/src/index.ts

```diff
@@ -388,9 +394,36 @@ export default class AwsS3Multipart<
     return this.#client
   }
 
+  #setClient(opts?: Partial<AwsS3MultipartOptions<M, B>>) {
+    if (
+      opts == null ||
+      !(
+        'endpoint' in opts ||
+        'companionUrl' in opts ||
+        'companionHeaders' in opts ||
+        'companionCookiesRule' in opts ||
+        'companionKeysParams' in opts
```

Comment:
forgot about these, sounds quite awkward if you want to send headers along to your own backend and you have to write `endpoint` and `companionHeaders`. Do we have to rename all then?

Reply:
The other awkward thing would be if one could not share the options with remote sources plugin, I’m not sure how to approach this.

Reply:
But this is not a remote source plugin, so there is no problem of inconsistency as this is an uploader? I think we can safely change this.

Reply:
Don't you think there are users that use remote sources with AWS plugin?

Reply:
It's possible, but then they have to just `endpoint`, `headers`, `cookiesRule` here. I don't think it matters. Note that what people have to do to with xhr and tus, so if anything it just aligns it more.